File: wheystore/wheystore.Web/Modules/SanPhamDB/Review/ReviewColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.Review")]
[BasedOnRow(typeof(ReviewRow), CheckNames = true)]
public class ReviewColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int SanPhamId { get; set; }
    public int KhachHangId { get; set; }
    public int UserId { get; set; }
    public short Rating { get; set; }
    [EditLink]
    public string TieuDe { get; set; }
    public string NoiDung { get; set; }
    public string AnhReview { get; set; }
    public bool DaDuyet { get; set; }
    public int SoLanHuuIch { get; set; }
    public bool BiAn { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ReviewColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ReviewRow } from "./ReviewRow";

export interface ReviewColumns {
    Id: Column<ReviewRow>;
    SanPhamId: Column<ReviewRow>;
    KhachHangId: Column<ReviewRow>;
    UserId: Column<ReviewRow>;
    Rating: Column<ReviewRow>;
    TieuDe: Column<ReviewRow>;
    NoiDung: Column<ReviewRow>;
    AnhReview: Column<ReviewRow>;
    DaDuyet: Column<ReviewRow>;
    SoLanHuuIch: Column<ReviewRow>;
    BiAn: Column<ReviewRow>;
    CreatedAt: Column<ReviewRow>;
    UpdatedAt: Column<ReviewRow>;
}

export class ReviewColumns extends ColumnsBase<ReviewRow> {
    static readonly columnsKey = 'SanPhamDB.Review';
    static readonly Fields = fieldsProxy<ReviewColumns>();
}